core/common/src/main/scala/net/liftweb/common/Box.scala

```diff
@@ -456,6 +456,13 @@ sealed abstract class Box[+A] extends Product with Serializable{
    */
   def flatMap[B](f: A => Box[B]): Box[B] = Empty
 
+  def flatten[B](implicit ev: A <:< Box[B]): Box[B] = this match {
```

Comment:
It does indeed serve as evidence, but we also *use* that evidence, so it is not there solely as evidence. I prefer the collections framework's approach in this case, personally.